cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/DistributedMapReduceTaskContextProvider.java

```diff
@@ -117,6 +117,6 @@ protected void shutDown() throws Exception {
   }
 
   private static Injector createInjector(CConfiguration cConf, Configuration hConf) {
-    return Guice.createInjector(new DistributedProgramRunnableModule(cConf, hConf).createModule());
+    return Guice.createInjector(new DistributedProgramRunnableModule(cConf, hConf).createModule(null, null));
```

Comment:
That means, if a partition is created from a MapReduce task, then the program id and principal are not passed along? Seems wrong to me...